### ISSUE TITLE
performance issue

### ISSUE THREAD

**Issue by manishorkus** [jacomyal/sigma.js#967]
I am using sigmajs to render some large graphs (30K edges). When I export sigmajs package from Gephi it works great (it uses an older version of sigmajs which is bundled with gephi). However when I load the same graph in the latest version of sigmajs the performance is much slower. Are there some specific flags enabled in the version of sigmajs that ships with gephi or is the new version of sigmajs slower as compared to the older version (v0.1).

**Comment by manishorkus** [comment]
Specifically I am seeing performance issue in zoom and unzoom when I see a stutter and delay

**Comment by Yomguithereal** [comment]
Are you using a mac with retina display?

**Comment by manishorkus** [comment]
Yes I am using a mac with retina but I am connected to a monitor which is only 1080p. Also the older version of sigmajs has no issues with rendering or zooming.

**Comment by Yomguithereal** [comment]
This is most probably not a performance issue but rather a mouse wheel inertia issue we have on mac browsers. Is panning the camera laggy?

**Comment by manishorkus** [comment]
Panning is fine. But the main issue is zooming. Is there a fix for the mouse wheel inertia issue that I can try?

**Comment by Yomguithereal** [comment]
I can't think of a fix that you could apply on your end. However next version of the lib (see v2) does not have this issue. I can try to either backport the fix or, if you feel adventurous, I can help you try v2 which is not completely stable nor documented. But it really depends on what features you need.

**Comment by manishorkus** [comment]
I am up for trying v2. Is there an example you can point to which uses v2 to create a basic graph. I am building a graph from json network data coming from gephi. For features I need ability to zoom, search and interact with the graph nodes. Layout is happening in gephi. Thanks a lot!

**Comment by Yomguithereal** [comment]
Check out the v2 branch `examples` folder for some examples. [This](https://github.com/jacomyal/sigma.js/blob/v2/examples/basic.js) one, for instance. You can install the latest dev version using npm:

```
npm install sigma@alpha
```

**Comment by Yomguithereal** [comment]
Note that it's using [graphology](https://graphology.github.io/) to represent the graph's data as the graph model & sigma are now separated.

**Comment by manishorkus** [comment]
I did try the v2 and the issue is fixed. However there seem to be other bugs in it as of now. Is it possible to give a backport for this to the stable version? Thx.

**Comment by Yomguithereal** [comment]
> However there seem to be other bugs in it as of now.

Such as?

> Is it possible to give a backport for this to the stable version?

I need to speak of this with @jacomyal first.

**Comment by manishorkus** [comment]
Couple of issues in V2 that I am seeing. One is that if you have labels on nodes and labels are set to show on zoom, at any given zoom level some nodes show labels while others don't even though all nodes are of same size. Another issue is that I have 15K edges and when I hover over a node I am highlighting the edges on that node, however the rendering of the highlighted edges is getting clobberred by the other edges drawn in the graph. If these are known issues please let me know if fixes are planned. Also I am using the v2 tag on the github repo and assuming that is the one i should be using.

Also if I was to backport the fix for mouse wheel inertial to 1.0 myself - can you point me to the code in 1.0 where the bug is and where in 2.0 it is getting fixed.

thx a lot for your help

**Comment by Yomguithereal** [comment]
> One is that if you have labels on nodes and labels are set to show on zoom, at any given zoom level some nodes show labels while others don't even though all nodes are of same size.

This is not a bug. This is by design. We are experimenting with a way to display a maximum of important labels on screen without saturating the screen if, say, the nodes all have the same size. There is a label selection algorithm which try to display the most important labels and dispatch them in space along a grid.

> Another issue is that I have 15K edges and when I hover over a node I am highlighting the edges on that node, however the rendering of the highlighted edges is getting clobberred by the other edges drawn in the graph.

You would also have this issue with v1. But in v2, you can enable z-index in the renderer settings and update edges/nodes z property to put the highlighted one into focus.

**Comment by manishorkus** [comment]
After setting zindex my edges are still going under other nodes.

Also how do I iterate through all nodes or edges in the graph in v2? For example if you click on a node I only want to show that node and its connected nodes. On doubleclick I want to show everything again. How do I accomplish that in v2?

**Comment by manishorkus** [comment]
Any chance I can get the fix for mac mouse inertia on the main stable release?

**Comment by bonyjo** [comment]
@Yomguithereal 
I am also facing a similar issue with scroll. For trying out the v2 can you please point me to some examples of v2.

**Comment by Yomguithereal** [comment]
Check out the `examples` folder of the `v2` branch.

**Comment by bonyjo** [comment]
@Yomguithereal  Thanks for the help. 
Quick question though does v2 support edges with an arrow and for version v1.2.1 we have some custom plugin function can the same be achieved here.

Thanks for the help in advance and I am a bit new in this area and trying to figure out stuff.

**Comment by Yomguithereal** [comment]
It kinda does but it is not straightforward to do so at the moment unfortunately.

**Comment by bonyjo** [comment]
@Yomguithereal  Thanks for the response. 
Is there a way this fix can be ported to Sigma main release?

**Comment by Yomguithereal** [comment]
It was at one point but then Apple decided again to mess things up. The only viable solution is to linearize the camera's easing. But I am unsure whether you can do it straightforwardly on v1.

Try changing [this](https://github.com/jacomyal/sigma.js/blob/master/src/captors/sigma.captors.mouse.js#L173) option to `linear`.
